### PR TITLE
migrate sqlite db users table to a per-maildir "password" file

### PR DESCRIPTION
**Pull request by hpk42**
- [x] set/read passwords from a "password" file in maildir 
- [x] set/read passwords for echobot from a special password file created by the echobot 
- [x] track last_login for a user as the "mtime" of the password file 
- [x] migration script for sqlite database to files (moving the old passdb.sqlite to passdb.sqlite.old)
- [x] new "User" object for manipulating passwords and last-login timestamp for a user with tests

supersedes #354  (this PR is based on it, but it's better to merge it all in one go, to not have partial db->file migrations)

### CODE REVIEW

**Comment by missytake** [comment]
Can you maybe change the HEAD this PR is to be merged into to #354? It would be easier to review then - first we could review #354, then this one, then merge this, and then (possibly without further review) #354. Breaks it up into smaller chunks

**Comment by hpk42** [comment]
> Can you maybe change the HEAD this PR is to be merged into to #354? It would be easier to review then - first we could review #354, then this one, then merge this, and then (possibly without further review) #354. Breaks it up into smaller chunks

I re-targeted the branch to merge with hpk/last-login which was already reviewed. 
Note btw, that even though there is a large "migrate_from_db_to_maildir" script with a large test, 
this PR cuts out lines because all the sqlite-handling is gone.

File: chatmaild/src/chatmaild/tests/test_delete_inactive_users.py [hunk 1]
```diff
@@ -1,49 +1,36 @@
...
-    userdir.mkdir()
-    write_last_login_to_userdir(userdir, timestamp=100000)
-    assert get_last_login_from_userdir(userdir) == 86400
+def test_login_timestamps(example_config):
+    testaddr = "someuser@chat.example.org"
+    user = example_config.get_user(testaddr)
 
-    write_last_login_to_userdir(userdir, timestamp=200000)
-    assert get_last_login_from_userdir(userdir) == 86400 * 2
+    user.set_password("1l2k3j1l2k3j123")
```
[the remark this answers is not in the document]
Comment (hpk42) [reply chatmaild/src/chatmaild/tests/test_delete_inactive_users.py:12]: added a comment

File: chatmaild/src/chatmaild/migrate_db.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+
+def migrate_from_db_to_maildir(config, chunking=10000):
+    path = config.passdb_path
+    if not path.exists():
+        return
+
+    all_rows = get_all_rows(path)
+
+    # don't transfer special/CI accounts (IOLO)
```
[the remark this answers is not in the document]
Comment (hpk42) [reply chatmaild/src/chatmaild/migrate_db.py]: removed it, it's not really YOLO anyway.

[the remark this answers is not in the document]
Comment (missytake) [reply chatmaild/src/chatmaild/migrate_db.py]: xD I thought this comment was about IO

**Pull request merged**